### PR TITLE
Fix logic error around postgresql_clusters in generators

### PR DESCRIPTION
**Pull request by avaczi**
In hindsight, it is not surprising there was no database discovery ever.

### CODE REVIEW

File: zmon_agent/discovery/kubernetes/cluster.py [hunk 1]
```diff
@@ -109,14 +109,14 @@ def get_entities(self) -> list:
             self.kube_client, self.cluster_id, self.alias, self.environment, self.region, self.infrastructure_account,
             namespace=self.namespace)
 
-        postgresql_cluster_entities = get_postgresql_clusters(
+        postgresql_cluster_entities, pce = itertools.tee(get_postgresql_clusters(
```
Comment (a1exsh) [inline zmon_agent/discovery/kubernetes/cluster.py:112]: Is this a more fancy way to say `list(...)`?

Comment (avaczi) [reply zmon_agent/discovery/kubernetes/cluster.py:112]: This keeps the generatorness of the generator (well, both of them), so we don't have a mix of list and generator based logic.

Comment (a1exsh) [reply zmon_agent/discovery/kubernetes/cluster.py:112]: Yeah, I see the extra "smartness" now...

**Comment by a1exsh** [comment]
@avaczi Could you please update the branch by merging master?

**Comment by mohabusama** [comment]
@avaczi I still see conflicts in cluster.py

**Comment by avaczi** [comment]
:+1:

**Comment by mohabusama** [comment]
👍

**Pull request merged**